### PR TITLE
improve js guessing

### PR DESCRIPTION
**Pull request by cdmistman**
- the query now handles dynamic imports
- the query now properly rejects `require` calls that don't have exactly 1 argument
- when getting the results of the query, always *just* get the `@import` tag

### CODE REVIEW

File: internal/backends/nodejs/grab.go [hunk 1]
```diff
@@ -105,24 +106,18 @@ func parseFile(contents []byte, results chan parseResult) {
...
+			if query.CaptureNameForId(capture.Index) == "import" {
+				importPath = capture.Node.Content(contents)
+				break
 			}
+		}
 
-			importPath = match.Captures[1].Node.Content(contents)
+		if importPath == "" {
+			// shouldn't happen, with the way the query is written and handled.
+			util.Die("Failed to parse import path")
```
Comment (masad-frost) [inline internal/backends/nodejs/grab.go:118]: I feel like we don't have enough observability to react to these effectively, but maybe we fixed that

File: test-suite/templates/guess/js/import
Comment (masad-frost) [inline test-suite/templates/guess/js/import:1]: Good to call this dynamici-mport

**Review by masad-frost — APPROVED** [review]
Nice

File: internal/backends/nodejs/grab.go (hunk 1 above)
Comment (cdmistman) [reply internal/backends/nodejs/grab.go:118]: @ryantm has a WIP branch adding o11y to upm

**Pull request merged**